### PR TITLE
fix: bg color for selected option in multiselect widget

### LINKED ISSUE #15493: [Bug]: Selected option(s) is not highlighted in a MultiSelect widget
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

After selecting any option(s) in the MultiSelect widget when we click on the dropdown again, the selected options are not highlighted. This behaviour is inconsistent from other Select widgets.

https://www.loom.com/share/ba791c0e43844d6e93060a90a097cd0e

### Steps To Reproduce

1. D&D a MultiSelect widget.
2. Select any option(s).
3. Click on the dropdown again.

**Expected:** The selected options should be highlighted.
**Actual:** The selected options are not highlighted.

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by SaiCharanChetpelly31**
Fixes #15493 
Added bg color styles to selected option(s)
Added cypress test.

## Summary by CodeRabbit

- **New Features**
	- Introduced automated tests for the MultiSelect widget to validate functionality and appearance.
  
- **Enhancements**
	- Improved the styling of selected options in the MultiSelect widget for better visual feedback, including dynamic background colors and hover effects.
  
- **Tests**
	- Added a comprehensive test suite for the MultiSelectWidget to ensure correct rendering and user interaction behavior.

### CODE REVIEW

**Comment by sagar-qa007** [comment]
@SaiCharanChetpelly31, We have seen there is change in src with business logic. This logic can be tested with Jest unit tests or React testing library. Cypress is redundant here.

**Comment by SaiCharanChetpelly31** [comment]
> @SaiCharanChetpelly31, We have seen there is change in src with business logic. This logic can be tested with Jest unit tests or React testing library. Cypress is redundant here.

@sagar-qa007 The bug involves adding a background color for a selected item. I wrote a Cypress test for it since it pertains to CSS. I didn't add any extra logic to the code; I simply added CSS for the selected option in the `index.styles.tsx` file.

Regarding the need for a new Jest test case for this bug fix, the Appsmith team hasn't written any Jest test cases yet. Is Cypress testing sufficient for this fix, or do I need to write Jest test cases as well? 

Thanks in advance. Looking forward to your reply.

**Comment by SaiCharanChetpelly31** [comment]
Hi @sagar-qa007 @ApekshaBhosale @ramsaptami Can you please take this pr forward for review?
Thank you.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Multiselect/MultiSelect6_spec.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  ]`;
+describe(
+  "Multi Select widget Tests",
+  { tags: ["@tag.Widget", "@tag.Multiselect"] },
+  function () {
+    before(() => {
+      entityExplorer.DragDropWidgetNVerify("multiselectwidgetv2", 550, 100);
+    });
+
+    it("1. Verify background color for selected option", () => {
```
Comment (sagar-qa007) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Multiselect/MultiSelect6_spec.ts:21]: Could you please help me understand the necessity of Cypress test cases in this context? I believe unit testing might be sufficient in this scenario.
Your insights would be appreciated.

**Review by sagar-qa007 — CHANGES_REQUESTED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Multiselect/MultiSelect6_spec.ts (hunk 1 above)
Comment (SaiCharanChetpelly31) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Multiselect/MultiSelect6_spec.ts:21]: @sagar-qa007 I have written unit tests for this PR.
I will remove cypress test code.

**Pull request closed without merging**